### DISCUSSION TITLE
Sonoff TH16 with DS18B20 as thermostat

### DISCUSSION THREAD

**Discussion by Paulxp** [arachnetech/homebridge-mqttthing#527] [Q&A]
Hi guys,
i use sonoff th16 with ds18b20 as thermostat in my flat. Tasmota with mqtt is installed on this device and i have lot of basic switches with this plugin. Works great. But need help with setup as mqtt thermostat

At this moment works fine this http thermostat plugin https://www.npmjs.com/package/homebridge-th16ermostat but want to replace it to mqttthing thermostat. http thermostat can't resolve status after homebridge restart. Everytime after HB restarts, i have to set temperature. Any template for config on mqttthing?

**Comment by HolgerAusB** [comment]
I don't know Tasmota. Can you send your temperature to the mqtt-broker with the retain-flag? So the broker wil cache/remember this value. As soon as a mqttthing (re)starts its subscription, it will get the last cached value. I use mosquitto_pub with -r option.

**Comment by rbswift** [comment]
Publishing with retain is one option but that means mqttthing could possibly get a very stale, outdated temperature value if the th16 has not published for a while for some reason.
Another option is to set the startpub topic to 'cmnd/devicename/tele' to request an immediate tele message including the Temperature on startup.
But I prefer to rely on the TelePeriod command on tasmota to send updates every 20 seconds which I think is the default when sensors are attached.

**Comment by rbswift** [comment]
This is my config but I use a BME280 instead of a DS18B20

```
        {
            "accessory": "mqttthing",
            "type": "temperatureSensor",
            "name": "Bathroom Temperature",
            "topics": {
                "getCurrentTemperature": {
                    "topic": "tele/sonoffbathroom/SENSOR",
                    "apply": "return (Math.round(JSON.parse(message).BME280.Temperature*2)/2);"
                }
            }
        },
```

**Reply by Paulxp** [reply]
Hi,
configuration Sonoff TH16 with ds18b20 as temperature sensor works fine, but i would like to set it as thermostat. [Thermostat](https://i.redd.it/6h75fb8zl7231.jpg) in homekit
